### ISSUE TITLE
Unable to update a value once you have set it once

### ISSUE THREAD

**Issue by andrewsnowden** [netbeast/redux-persist-react-native-fs#1]
Hey,

So we ported some of this to use on Android phones but this line seems to be wrong:

```
const nextTree = { [key]: JSON.parse(value), ...tree }
```

If you already have the key in your existing tree, then it will just overwrite your new value. So you can write a value once, but then you can never update it again.

Suggestion is to use something like

```
const nextTree = Object.assign({}, tree, {key: JSON.parse(value)})
```

Or even a simple assign if you don't mind it being mutated.

**Comment by jsdario** [comment]
Sorry, this was a major mistake. We were using a local copy of this file that has the correct expressions.
In ES6/7 syntax it would be read as:

```
const nextTree = { ...tree, [key]: value }
```

I'll create a PR and submit for you to review, if it is alright.
Thanks a lot!

**Comment by jsdario** [comment]
I directly updated it into master, you can check the changes [here](https://github.com/netbeast/redux-persist-react-native-fs/commit/605033e84da9ff151522cb8e24ecda0efde615eb)
I also published it to npm.

**Comment by HaswinVidanage** [comment]
@jsdario is this issue fixed?

**Comment by jsdario** [comment]
Yes, but this project has other issues, depending on the version of redux-persist that you are using. In v4 at least, some parts of the state can be overwritten due to racing conditions on the use of redux-persist.
